codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java

```diff
@@ -1613,6 +1678,131 @@ private void buildAbiFunction(
+                List.class,
+                FunctionReturnDecoder.class);
+        methodBuilder.addStatement(
+                "$N $N = new $N()",
+                responseClassName,
+                Strings.lowercaseFirstLetter(responseClassName),
+                responseClassName
+        );
+
+        int parameterIndex = 0;
+        for (AbiDefinition.NamedType namedType : outputs) {
+            parameterIndex += 1;
+            String parameterName = namedType.getName();
+            if ("".equals(parameterName)) {
+                parameterName = "return".concat(String.valueOf(parameterIndex));
+            }
+            if (useNativeJavaTypes) {
+                methodBuilder.addStatement("$N.$N = ($T) response.get($L).getValue()",
+                        Strings.lowercaseFirstLetter(responseClassName),
+                        parameterName,
+                        getNativeType(outputParameterTypes.get(parameterIndex-1)),
+                        parameterIndex-1
+                );
+            } else {
+                methodBuilder.addStatement("$N.$N = ($T) response.get($L)",
+                        Strings.lowercaseFirstLetter(responseClassName),
+                        parameterName,
+                        outputParameterTypes.get(parameterIndex-1),
+                        parameterIndex-1
+                );
+            }
+        }
+
+        methodBuilder.addStatement(
+                "return $N",
+                Strings.lowercaseFirstLetter(responseClassName)
+        );
+    }
+
+    TypeSpec buildFunctionResponseObject(
+            String className,
+            List<org.web3j.codegen.SolidityFunctionWrapper.NamedTypeName> outputParameters) {
+
+        TypeSpec.Builder builder =
+                TypeSpec.classBuilder(className).addModifiers(Modifier.PUBLIC, Modifier.STATIC);
+
+        int fieldIndex = 0;
+        for (org.web3j.codegen.SolidityFunctionWrapper.NamedTypeName namedType :
+                outputParameters) {
+            final TypeName typeName;
+            fieldIndex += 1;
+            if (namedType.getType().equals("tuple")) {
+                typeName = structClassNameMap.get(namedType.structIdentifier());
+            } else if (namedType.getType().startsWith("tuple")
+                    && namedType.getType().contains("[")) {
+                typeName = buildStructArrayTypeName(namedType.namedType, true);
+            } else {
+                if (useNativeJavaTypes) {
+                    typeName = getNativeType(namedType.typeName);
+                } else {
+                    typeName = namedType.typeName;
+                }
+            }
+            if ("".equals(namedType.getName())) {
+                builder.addField(typeName, "return".concat(String.valueOf(fieldIndex)), Modifier.PUBLIC);
+            } else {
+                builder.addField(typeName, namedType.getName(), Modifier.PUBLIC);
```

Comment:
will throw error here if function won't have parameter name
I see that you took the example from buildEventResponseObject method, take last merges and see the fix

Reply:
Applied changes as suggested